src/core/mac/mac_types.hpp

```diff
@@ -408,12 +408,18 @@ class Address
     Type mType; ///< The address type (Short, Extended, or none).
 };
 
+enum CryptoType
+{
+    kUseKeyLiterals = OT_KEY_TYPE_LITERAL_KEY,
+    kUseKeyRefs     = OT_KEY_TYPE_KEY_REF,
+};
+
 /**
  * This class represents a MAC key.
  *
  */
 OT_TOOL_PACKED_BEGIN
-class Key : public otMacKey, public Equatable<Key>, public Clearable<Key>
+class Key : public otMacKeyMaterial, public Equatable<Key>, public Clearable<Key>
 {
```

Comment:
One suggestion here:

How about renaming this class here to be `KeyMatreial` instead of `Key`?
- This way the the C++ class name `Mac::KeyMaterial` mirrors the `otMacKeyMateral` C platform type name
- It also highlights the notion that it is a key material (and not necessarily literal key).

With this rename, we can then keep using `Mac::Key` class name as before (as a subclass of `otMacKey` to represent a literal key (the 16 byte buffer) and then use in `KeyManager` when deriving the literal keys (e.g. in `HashKey` etc).

Reply:
I can rename it, but we can't have `Key` class as it will not be used anywhere, as I mentioned above.